### PR TITLE
Add basic text input

### PR DESCRIPTION
**Pull request by lessp**
Adds a basic text input component. This adds `minttea` as a dependency to `leaves`, see this comment for why https://github.com/leostera/minttea/issues/5#issuecomment-1868999916, but in short it's because we're then able to simply forward the key event to the text-inputs `update` function.

It draws some inspiration from https://github.com/charmbracelet/bubbles/blob/master/textinput/textinput.go. It does however miss quite a few features. For example `placeholder`, multiple combined key-commands like `alt + left` and so forth. It should however be rather easy to expand the text input with these features, given that we're OK with the current approach.

## Video

https://github.com/leostera/minttea/assets/17602389/68eaaed1-7541-4abf-a5bd-f3d22af15810

### CODE REVIEW

File: leaves/text_input.mli [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     let text_input = Text_input.update text_input key
+   ]}
+*)
+
+val write : t -> string -> t
+val space : t -> t
+val backspace : t -> t
+val move_cursor : t -> [ `Character_backward | `Character_forward ] -> t
+val character_backward : t -> t
+val character_forward : t -> t
```
Comment (leostera) [inline leaves/text_input.mli:58]: **question**. Do we want these exposed so people can manually update the state?

File: leaves/text_input.mli [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  ?cursor_style:Spices.style ->
+  unit ->
+  t
+(**
+   Create a new {Text_input}. You can optionally provide styles for the text and/or the cursor.
+   
+   {[
+     let text_input = Text_input.make ~value:"Hello" ()
+   ]}
+*)
```
Comment (leostera) [inline leaves/text_input.mli:20]: Thanks for the interface docs 🫰🏼

File: leaves/text_input.mli [hunk 3]
```diff
@@ -0,0 +1,58 @@
+type t = {
```
Comment (leostera) [inline leaves/text_input.mli:1]: **fix**. let's make this abstract.

**Review by leostera — COMMENTED** [review]
Thanks @lessp! ✨ This is a lovely addition. I have just one tiny fix to make the type of Text_input.t abstract (like we did for Fps and Sprite). Then it's a LGTM from me! 🚀

File: leaves/text_input.mli (hunk 1 above)
Comment (lessp) [reply leaves/text_input.mli:58]: Not sure, so perhaps let's hide them until we need them?

File: leaves/text_input.ml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    | `Character_backward when t.cursor > 0 -> t.cursor - 1
+    | `Character_forward when t.cursor < String.length t.value -> t.cursor + 1
+    | _ -> t.cursor
+  in
+  { t with cursor = new_cursor }
+
+let character_backward t = move_cursor t `Character_backward
+let character_forward t = move_cursor t `Character_forward
+
+let update t (key : Minttea.Event.key) =
```
Comment (lessp) [inline leaves/text_input.ml:81]: @leostera, do you think this should be `Minttea.Event.t` in case we need something from the other events, or good as is?

Comment (lessp) [reply leaves/text_input.ml:81]: E.g. cursor blinking, potentially modifier keys (although might make more sense to expose this from `Event.key`), and so forth.

Comment (lessp) [reply leaves/text_input.ml:81]: Addressed: [1f0f4aa](https://github.com/leostera/minttea/pull/16/commits/1f0f4aa9939b1ba15d872640312b5463bc7dc44b)

File: leaves/text_input.mli (hunk 3 above)
Comment (lessp) [reply leaves/text_input.mli:1]: In this case, should we rename `value` -> `text` for `val make : ~value: string -> ...`?

File: leaves/text_input.mli (hunk 1 above)
Comment (lessp) [reply leaves/text_input.mli:58]: Removed in: [1f0f4aa](https://github.com/leostera/minttea/pull/16/commits/1f0f4aa9939b1ba15d872640312b5463bc7dc44b)

**Comment by lessp** [comment]
FWIW, I also added `prompt` with a default of `> `. See [1f0f4aa](https://github.com/leostera/minttea/pull/16/commits/1f0f4aa9939b1ba15d872640312b5463bc7dc44b)

File: leaves/text_input.ml (hunk 4 above)
Comment (lessp) [reply leaves/text_input.ml:81]: Ehh. Looks like I didn't include all changes.

File: leaves/text_input.mli (hunk 3 above)
Comment (lessp) [reply leaves/text_input.mli]: ... alternatively simply remove the named argument for value

Comment (lessp) [reply leaves/text_input.mli]: I did the latter: [7f96876](https://github.com/leostera/minttea/pull/16/commits/7f96876cd5e96fa2efea3a79bc9492e94e192d65)

**Comment by lessp** [comment]
Ok, think we're good to go post review.

**Review by leostera — APPROVED** [review]
LGTM! 🚀 Thanks again for the contribution 🤩

**Pull request merged**